client/extensions/woocommerce/index.js

```diff
@@ -151,15 +155,23 @@ function addStorePage( storePage, storeNavigation ) {
 	} );
 }
 
-function createStoreNavigation( context, next ) {
-	const state = context.store.getState();
-	const selectedSite = getSelectedSite( state );
+function mapStateToProps( state ) {
+	let siteId = getSelectedSiteId( state );
+	if ( ! siteId && ( 1 === getCurrentUserSiteCount( state ) ) ) {
+		siteId = getPrimarySiteId( state );
+	}
+	const site = getSite( state, siteId );
 
+	return {
+		site
+	};
+}
+
+function createStoreNavigation( context, next ) {
 	renderWithReduxStore(
 		React.createElement(
-			StoreSidebar, {
+			connect( mapStateToProps )( StoreSidebar ), {
```

Comment:
Fixed in https://github.com/Automattic/wp-calypso/pull/14022/commits/57501ef4662e51dec3431a8778c6a7c588560785